### PR TITLE
Numpy compatible multinomial

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
numpy compatible multinomial

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
@haojin2 @reminisce

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review, Operator]

**Comment by reminisce** [comment]
Please add the python documentation to `numpy_op_doc.py` for this operator as we discussed.

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -86,3 +86,33 @@ def _np_zeros_like(a):
         Array of zeros with the same shape and type as `a`.
     """
     pass
+
+
+def _npi_multinomial(a):
```
Comment (stu1130) [inline python/mxnet/_numpy_op_doc.py]: @reminisce is here the right place:

Comment (reminisce) [reply python/mxnet/_numpy_op_doc.py]: Yes, thanks.

File: tests/python/unittest/test_numpy_ndarray.py [hunk 2]
```diff
@@ -625,6 +625,52 @@ def convert(num):
...
+            freq = mx.np.random.multinomial(experiements, pvals, size=size).asnumpy()
+            assert freq.size == 0
+    # check [] as pvals
+    for pvals in [[], ()]:
+        freq = mx.np.random.multinomial(experiements, pvals).asnumpy()
+        assert freq.size == 0
+        for size in sizes:
+            freq = mx.np.random.multinomial(experiements, pvals, size=size).asnumpy()
+            assert freq.size == 0
+
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_ndarray.py]: nit: get rid of the extra blank line.

File: python/mxnet/ndarray/numpy/random.py [hunk 3]
```diff
@@ -17,11 +17,14 @@
...
 """Namespace for operators used in Gluon dispatched by F=ndarray."""
 from __future__ import absolute_import
+import numpy as np
 from ...base import numeric_types
 from ...context import current_context
+from ..ndarray import NDArray
 from . import _internal as _npi
 
-__all__ = ['uniform', 'normal']
+
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/random.py]: nit: no need for the extra blank line.

File: python/mxnet/numpy/random.py [hunk 4]
```diff
@@ -98,3 +98,32 @@ def normal(loc=0.0, scale=1.0, size=None, **kwargs):
     This function currently does not support ``loc`` and ``scale`` as ndarrays.
     """
     return _mx_nd_np.random.normal(loc, scale, size, **kwargs)
+
```
Comment (haojin2) [inline python/mxnet/numpy/random.py]: nit: one more blank line here.

File: tests/python/unittest/test_numpy_ndarray.py (hunk 2 above)
Comment (stu1130) [reply tests/python/unittest/test_numpy_ndarray.py]: done!

File: python/mxnet/ndarray/numpy/random.py (hunk 3 above)
Comment (stu1130) [reply python/mxnet/ndarray/numpy/random.py]: done!

File: python/mxnet/numpy/random.py (hunk 4 above)
Comment (stu1130) [reply python/mxnet/numpy/random.py]: done!

**Pull request merged**